src/packet/packetlist.js

```diff
@@ -64,6 +64,9 @@ class PacketList extends Array {
    * @async
    */
   async read(bytes, allowedPackets, config = defaultConfig) {
+    if (config.additionalAllowedPackets && config.additionalAllowedPackets.length) {
```

Comment:
I don't think the check is needed? `constructAllowedPackets` will simply return an empty map on `[]` input.